test/e2e/e2e_test.go

```diff
@@ -68,20 +68,46 @@ func TestOperator(t *testing.T) {
 	// Deploying the controller-manager
 	_, err = kind("make", "deploy", "install", fmt.Sprintf("IMG=%s", img))
 	require.NoError(t, err)
+	// Accept the EULA and set the PID
+	_, err = kind("kubectl", "set", "env",
+		"-n", nsController, controller,
+		"MSSQL_ACCEPT_EULA=Y", "MSSQL_PID=Developer")
+	require.NoError(t, err)
 	// Restarting the controller-manager, to ensure the latest image is used
 	_, err = kind("kubectl", "rollout", "restart",
 		"-n", nsController, controller)
 	require.NoError(t, err)
 	// Waiting for the controller-manager to be ready
 	_, err = kind("kubectl", "rollout", "status",
-		"-n", nsController, controller, "--timeout", "2m")
+		"-n", nsController, controller,
+		"--timeout", "2m")
 	require.NoError(t, err)
+	var controllerPod string
+	for range 5 {
+		// Wait for the old pods to be deleted
+		<-time.After(time.Second)
+		// Getting the controller-manager pod name
+		output, err := kind("kubectl", "wait", "pod",
+			"-n", nsController,
+			"-l", "control-plane=controller-manager",
+			"--for", "condition=Ready",
+			"-o", "go-template",
+			"--template", "{{.metadata.name}}",
+		)
+		require.NoError(t, err)
+		pods := strings.Split(output, "\n")
+		if len(pods) == 1 {
+			controllerPod = pods[0]
```

Comment:
Can this work with kubectl wait?

Reply:
No. :( Both old and new old still in the running state, it cause conflicts then pick the pod in ternimation process